octopusdeploy_framework/resource_deployment_freeze_test.go

```diff
@@ -0,0 +1,130 @@
+package octopusdeploy_framework
+
+import (
+	"fmt"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/deploymentfreezes"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestNewDeploymentFreezeResource(t *testing.T) {
+	localName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	resourceName := "octopusdeploy_deployment_freeze." + localName
+	name := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	start := fmt.Sprintf("%d-11-21T06:30:00+10:00", time.Now().Year()+1)
+	end := fmt.Sprintf("%d-11-21T08:30:00+10:00", time.Now().Year()+1)
+	projectName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	environmentName1 := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	environmentName2 := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	spaceName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	projectGroupName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	lifecycleName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
+		CheckDestroy:             testDeploymentFreezeCheckDestroy,
+		PreCheck:                 func() { TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: ProtoV6ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Check: resource.ComposeTestCheckFunc(
+					testDeploymentFreezeExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "start", start),
+					resource.TestCheckResourceAttr(resourceName, "end", end)),
+				Config: testDeploymentFreezeBasic(localName, name, start, end, spaceName, []string{environmentName1}, projectName, projectGroupName, lifecycleName),
+			},
+		},
```

Comment:
Would appreciate if we run a 2nd step with an updated config so we can test the update function for the resource

Reply:
I put in an update as a second `resource.Test` block. I will move it into the same set and make the update do more changes